core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -391,6 +394,24 @@ public void rename(AlluxioURI src, AlluxioURI dst, RenameOptions options)
     }
   }
 
+  @Override
+  public void setAcl(AlluxioURI path, SetAclAction action, List<AclEntry> entries,
+      SetAclOptions options) throws FileDoesNotExistException, IOException, AlluxioException {
+    FileSystemMasterClient masterClient = mFileSystemContext.acquireMasterClient();
+    try {
+      masterClient.setAcl(path, action, entries, options);
+      LOG.debug("Set ACL for {}, entries: {} options: {}", path.getPath(), entries, options);
+    } catch (NotFoundException e) {
+      throw new FileDoesNotExistException(e.getMessage());
+    } catch (UnavailableException e) {
+      throw e;
+    } catch (AlluxioStatusException e) {
+      throw e.toAlluxioException();
```

Comment:
Yeah I was just curious about the difference, I think it might have to do with that we are relying on `UnavailableException` to retry to client request.